aptos-move/aptos-vm-types/src/abstract_write_op.rs

```diff
@@ -89,7 +86,10 @@ impl AbstractResourceWriteOp {
                 metadata_op: write_op,
                 ..
             }) => write_op.get_metadata_mut(),
-            InPlaceDelayedFieldChange(_) | ResourceGroupInPlaceDelayedFieldChange(_) => None,
+            InPlaceDelayedFieldChange(_) | ResourceGroupInPlaceDelayedFieldChange(_) => {
+                // FIXME(aldenhu): confirm this is correct
+                unreachable!("No metadata for delayed field changes")
```

Comment:
we are iterating over all writes, this is definitely reachable - that's why it was an option?

"InPlaceDelayedFieldChange" means we are doing a write to that state item, it is just that only write is from Aggregator inside it, there are no other updates. 

and so the actual io write needs to be charged, it's just that state will stay the same size

Reply:
As discussed, made the change to let `InPlaceDelayedFieldChange` carry the metadata as well.